### PR TITLE
#526 : Release binaries for Registry Tools

### PR DESCRIPTION
**Pull request by giteshk**
1. Added a release workflow to generate Registry tool binaries (#526)
2. Added a version to the registry tools command (#535)

### CODE REVIEW

**Comment by giteshk** [comment]
Here is a sample release with the binaries looks like:

https://github.com/giteshk-org/registry/releases/tag/v0.3.13

**Review by theganyo — APPROVED** [review]
LGTM

You may want to update the README to let people know that a binary is available in the releases instead of having to build.

Also, you may want to mention somewhere that on MacOS you need to unquaranteen the binary before you can run it. This can be done via `xattr -d com.apple.quarantine registry`.

**Comment by timburks** [comment]
Are we following an existing convention for naming the release files?

If there's no strong precedent, I'd like to follow the convention used by [protoc releases](https://github.com/protocolbuffers/protobuf/releases)
```
registry-3.2.0-osx-x86_64.zip
registry-3.2.0-linux-x86_64.zip
registry-3.2.0-win64.zip
```

**Comment by theganyo** [comment]
No particular precedent, I didn't override the goreleaser default. 
(FWIW, it looks like [Istio](https://github.com/istio/istio/releases/tag/1.13.3) uses yet another slightly different approach as well, eg. `istio-1.13.3-linux-amd64.tar.gz`)

**Comment by timburks** [comment]
I see - goreleaser is the precedent then - anything built with goreleaser defaults would use these conventions? If so, this LGTM.

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - amd64
+  env:
+  - CGO_ENABLED=0
+  ldflags:
+  - -s -w -X "github.com/apigee/registry/cmd/registry/cmd.Version={{.Env.RELEASE_VERSION}}"
+
+archives:
+- id: default
+  format: tar.gz
+  replacements:
```
Comment (timburks) [inline .goreleaser.yml:34]: For the sake of predictability, I would rather not make these replacements (going with the defaults seems more aligned with what other projects would do)

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    darwin: macOS
+    amd64: 64-bit
```
Comment (timburks) [inline .goreleaser.yml:36]: For the sake of predictability, I would rather not make these replacements (going with the defaults seems more aligned with what other projects would do)

File: auth/HOSTED.bat [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (timburks) [inline auth/HOSTED.bat:1]: I think this needs a license header.

File: cmd/registry/cmd/root.go [hunk 4]
```diff
@@ -32,10 +32,13 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var Version = "dev"
```
Comment (timburks) [inline cmd/registry/cmd/root.go:35]: Is this being overridden somewhere in the build process?

Comment (timburks) [reply cmd/registry/cmd/root.go:35]: Here? 
```
ldflags:
  - -s -w -X "github.com/apigee/registry/cmd/registry/cmd.Version={{.Env.RELEASE_VERSION}}"
```

Comment (giteshk) [reply cmd/registry/cmd/root.go:35]: Yes. we are pulling the RELEASE_VERSION from the tag name in the release.yml workflow file

Comment (seaneganx) [reply cmd/registry/cmd/root.go:35]: I think we should add a comment to explain the purpose here.

File: auth/HOSTED.bat (hunk 3 above)
Comment (seaneganx) [reply auth/HOSTED.bat:1]: +1

**Review by seaneganx — APPROVED** [review]
LGTM after open comments are addressed.

File: cmd/registry/cmd/root.go (hunk 4 above)
Comment (timburks) [reply cmd/registry/cmd/root.go:35]: Agreed, when I first saw this in the code I thought it was unfinished.

Comment (giteshk) [reply cmd/registry/cmd/root.go:35]: comment added

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Prepare
+        id: prep
+        run: |
+          echo ::set-output name=version::${GITHUB_REF#refs/tags/}
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.17
```
Comment (timburks) [inline .github/workflows/release.yml:39]: Can you update this to 1.18 ?

File: .goreleaser.yml [hunk 6]
```diff
@@ -33,7 +33,6 @@ archives:
   format: tar.gz
   replacements:
     darwin: macOS
```
Comment (timburks) [inline .goreleaser.yml:35]: can we remove this also?

**Review by timburks — APPROVED** [review]

**Pull request merged**